File: shared/src/main/java/com/gcgenome/lims/data/Worklist.java
package com.gcgenome.lims.data;

import jsinterop.annotations.*;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@JsType(isNative=true, namespace= JsPackage.GLOBAL, name="Object")
@Setter(onMethod_= {@JsOverlay, @JsIgnore})
@Getter(onMethod_= {@JsOverlay, @JsIgnore})
@Accessors(fluent=true)
public final class Worklist {
    private String id;
    private String title;
    @JsProperty(name="created_at")
    private String created;
    private String status;
    private String remark;
    private String serial;
    private String prefix;
    private Double idx;

    @JsOverlay
    @JsIgnore
    public Status status() {
        if (status == null) return null;
        return Status.valueOf(status);
    }
    @JsOverlay
    @JsIgnore
    public Worklist idx(Integer idx) {
        if (idx == null) this.idx = null;
        else this.idx = idx.doubleValue();
        return this;
    }

    public enum Status{
        NORMAL,         // 검증완료
        PRE_CREATE,     // Worklist 저장
        POST_CREATE,    // Worklist + Plate
        DISPOSAL,
        MERGED,
        SEQUENCING,
        ANALYZED,
        COMPLETE
    }
}

File: worklist-api/src/main/kotlin/com/gcgenome/lims/data/Worklist.kt
package com.gcgenome.lims.data

data class Worklist (
    val id: String = ""
) {
    var title:          String  = ""
    var createAt:       String  = ""
    var createBy:       String  = ""
    var status:         String  = ""
    var remark:         String  = ""
    var domain:         String  = ""
    var serial:         String  = ""
    var prefix:         String  = ""
    var idx:            Int     = -999999
    var lastModifyAt:   String  = ""
    var serializeBy:    String  = ""
}
